translations_parser/parser.py

```diff
@@ -0,0 +1,228 @@
+import logging
+import re
+from collections import defaultdict
+from dataclasses import dataclass
+from datetime import datetime
+from typing import List
+
+import yaml
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)s] %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+HEADER_RE = re.compile(r"(?<=\[)(?P<value>.+?)\] ")
+# fmt: off
+VALIDATION_RE = re.compile(
+    r"Ep\.[ :]+(?P<ep>\d+)[ :]+"
+    r"Up\.[ :]+(?P<up>\d+)[ :]+"
+    r"(?P<key>[\w-]+)[ :]+(?P<value>[\d\.]+)"
+)
+# fmt: on
+TRAINING_RE = re.compile(
+    r"Ep\.[ :]+(?P<epoch>\d+)[ :]+"
+    r"Up\.[ :]+(?P<up>\d+)[ :]+"
+    r"Sen\.[ :]+(?P<sen>[\d,]+)[ :]+"
+    r"Cost[ :]+(?P<cost>[\d.]+)[ :]+"
+    r"Time[ :]+(?P<time>[\d\.]+s)[ :]+"
+    r"(?P<rate>[\d\.]+ words\/s)[ :]+"
+    r"gNorm[ :]+(?P<gnorm>[\d\.]+)"
+)
+
+
+@dataclass
+class Training:
```

Comment:
```suggestion
class TrainingEpoch:
```

As it's only 1 epoch, and not the whole thing